### PR TITLE
Add ESM support

### PR DESCRIPTION
**Pull request by Codex-**

### CODE REVIEW

**Comment by darkobits** [comment]
Thank you for addressing this! 🙌

File: lib/loader.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
 import type { Loader } from "cosmiconfig";
 import { register, RegisterOptions } from "ts-node";
 
-import { TypeScriptCompileError } from "./typescript-compile-error";
+import { TypeScriptCompileError } from "./typescript-compile-error.js";
 
 export function TypeScriptLoader(options?: RegisterOptions): Loader {
   const tsNodeInstance = register({
     ...options,
     compilerOptions: { module: "commonjs" },
```
Comment (darkobits) [inline lib/loader.ts:9]: FYI as of version `8.2.0`, Cosmiconfig no longer requires transpiled configuration to be CJS; [they now accept pure ESM configuration files when using the asynchronous API](https://github.com/cosmiconfig/cosmiconfig/blob/main/CHANGELOG.md#820).

Comment (Codex-) [reply lib/loader.ts:9]: yeah haven't had another chance yet to finish this off, this and the usage of require/import might need some changes during compilation to generate the appropriate sources

Comment (Codex-) [reply lib/loader.ts]: @darkobits if you get a chance, this branch is now ready for testing :)

**Comment by Codex-** [comment]
Closed in favour of #106

**Pull request closed without merging**